doc/wiki/notes/ch4_inline.md

```diff
@@ -1,83 +1,86 @@
-The note documents the inlining strategy in CH4.
-
-The goal of inlining is to avoid the function call overhead (especially
-for functions with many arguments). It is the most beneficial for the
-latency sensitive code path---communication related code path. For
-other codes like Init/Finalize and Comm/Win creation/destroy function
-inlining is less beneficial.
-
-The guidelines for inlining are as follow.
-
-1. PT2PT, Collective and RMA communication functions need to be inlined.
-   This also means any function that is used in this code path need to
-   be inlined as well. For example, request creation, AM fallback for
-   communication.
-
-2. Any function that does not fall in the first rule should not be
-   inlined.
-
-3. For a given component, some of its functions may need to be inlined
-   while others do not. For example, in ch4r_buf.h/.c the creation and
-   destroy of buf pool is only used in init/finalize code, therefore no
-   inlining is needed. But the functions for obtaining and returning
-   individual buffers are used in the communication code path,
-   therefore, inlining is required.
-
-Uninlining procedure for CH4 layer
-
-1. Create a src/*.c file, include `mpidimpl.h`.
-2. For functions need to be uninlined, move them to *.c and create
-   create declarations in the *.h file. Note that the functions
-   implemented the ADI do not need this as the prototype is already
-   declared in `mpidch4.h`.
-3. Remove `static inline` or `MPL_STATIC_INLINE_PREFIX` of the uninlined
-   functions (both declarations `mpidch4.h`  and definitions)
-4. For uninlined functions that is only used in the *.c file, make them
-   “private”. 1) remove their namespace prefix; 2) create declarations
-   in *.c file; 3) make them `static`.
-5. Remove the *.h file if they are empty. `ch4r_*.h` files are kept for
-   the decls
-6. Keep CVAR to the place where it is used.
-
-Uninlining procedure for netmod/shm layer
-
-1. Create *.c file, include `mpidimpl.h`.
-2. For functions need to be uninlined, move them to *.c. Rename from
-   `MPIDI_NM_*` to `MPIDI_OFI_*`. Create decls and #define in
+# CH4 Inline Mechanism
+
+This page discusses the inlining strategy in CH4.
+
+The goal of inlining is to avoid the function call overhead, especially for
+functions with many arguments. It is the most beneficial for the latency
+sensitive code path, communication. For other functions like, `Init/Finalize`,
+`Comm/Win`, `Create/Destroy`, inlining is less beneficial.
+
+## Guidelines For Inlining
+
+1. PT2PT, Collective and RMA communication functions need to be inlined. This
+   also means any function that is used in this code path needs to be inlined
+   as well. For example, request creation or AM fallback for communication.
+
+2. Any function that is not described by the first rule should not be inlined.
```

Comment:
Change "should not" to "need not to".

Reply:
is "...does not need to be inlined." ok?

Reply:
Sure.

Reply:
Fixed.